Andreno Juanda

Andrenoj11 opened the pull request.

Andrenoj11 committed:
challenge api doc dan rest api done

Sursev07 commented on server/routes/index.js:
```diff
@@ -0,0 +1,6 @@
+const todosRouter = require('./todos-router')
+
+router.use('/todos', todosRouter)
+
```
lanjut lagi untuk login dan register yah

Andrenoj11 committed:
nambahin autentikasi dan autorisasi

Andrenoj11 committed:
nambahin error handler

Andrenoj11 committed:
nambahin 3rd party

Andrenoj11 committed:
melengkapi api doc

Andrenoj11 committed:
rev

Andrenoj11 committed:
update typo

Andrenoj11 committed:
update api doc get todos news

Sursev07 commented on server/app.js:
```diff
@@ -0,0 +1,16 @@
+const port = process.env.POST || 3000
+const errorHandler = require('./middlewares/error-handler')
+
+app.use(express.urlencoded({ extended: false }))
```
cors jangan lupa yah

Sursev07 commented on server/helper/password.js:
```diff
@@ -0,0 +1,7 @@
+const bcrypt = require('bcryptjs')
+
+function comparePw(pw, hashPw){
```
kenapa ga sekalian hashpassorwd nya juga dipindah ke helper? 😄

Sursev07 commented on server/middlewares/authorization.js:
```diff
@@ -0,0 +1,27 @@
+        id: req.params.id
+    }})
+        .then(todo => {
+            if(todo.UserId == req.userLogIn.id){
```
kalau todo nya kosong bakal langsung error ke catch kan?
todo null
tapi kamu akses if(**todo.UserId**)

Sursev07 commented on server/middlewares/error-handler.js:
```diff
@@ -0,0 +1,19 @@
+module.exports = (err, req, res, next)=>{
```
naming file nya ga mending 
errorHandler ?

Andrenoj11 committed:
update jquery

Andrenoj11 committed:
update oauth google sign in

Andrenoj11 committed:
add features:
1. complete log in display in client
2. complete register display in client
3. complete fetching the guardian news api in client and display it

Andrenoj11 committed:
add feature updates data on todo list in client

Andrenoj11 committed:
hide update form in main page

Andrenoj11 committed:
change color button cancel in update todos list form

Andrenoj11 committed:
fix bug when log in from google api can not fetch data

Andrenoj11 committed:
deploy test

Andrenoj11 committed:
change ajax url

Andrenoj11 committed:
revisi all requirement in fancy todo
